homeassistant/components/islamic_prayer_times/__init__.py

```diff
@@ -116,60 +125,35 @@ async def async_schedule_future_update(self):
+
+    async def _async_update_data(self, *_) -> dict[str, datetime]:
         """Update sensors with new prayer times."""
         try:
             prayer_times = await self.hass.async_add_executor_job(
                 self.get_new_prayer_times
             )
-            self.available = True
-        except (exceptions.InvalidResponseError, ConnError):
-            self.available = False
-            _LOGGER.debug("Error retrieving prayer times")
-            async_call_later(self.hass, 60, self.async_update)
-            return
-
-        for prayer, time in prayer_times.items():
-            self.prayer_times_info[prayer] = dt_util.parse_datetime(
-                f"{dt_util.now().date()} {time}"
-            )
-        await self.async_schedule_future_update()
-
-        _LOGGER.debug("New prayer times retrieved. Updating sensors")
-        async_dispatcher_send(self.hass, DATA_UPDATED)
-
-    async def async_setup(self):
-        """Set up the Islamic prayer client."""
-        await self.async_add_options()
-
-        try:
-            await self.hass.async_add_executor_job(self.get_new_prayer_times)
         except (exceptions.InvalidResponseError, ConnError) as err:
-            raise ConfigEntryNotReady from err
-
-        await self.async_update()
-        self.config_entry.add_update_listener(self.async_options_updated)
-
-        self.hass.config_entries.async_setup_platforms(self.config_entry, PLATFORMS)
+            async_call_later(self.hass, 60, self.async_request_update)
+            raise UpdateFailed from err
 
-        return True
+        prayer_times_info: dict[str, datetime] = {}
+        for prayer, time in prayer_times.items():
+            if prayer_time := dt_util.parse_datetime(f"{dt_util.now().date()} {time}"):
+                prayer_times_info[prayer] = dt_util.as_utc(prayer_time)
 
-    async def async_add_options(self):
-        """Add options for entry."""
-        if not self.config_entry.options:
-            data = dict(self.config_entry.data)
-            calc_method = data.pop(CONF_CALC_METHOD, DEFAULT_CALC_METHOD)
+        await self.async_schedule_future_update(prayer_times_info["Midnight"])
+        return prayer_times_info
 
-            self.hass.config_entries.async_update_entry(
-                self.config_entry, data=data, options={CONF_CALC_METHOD: calc_method}
-            )
 
-    @staticmethod
-    async def async_options_updated(hass: HomeAssistant, entry: ConfigEntry) -> None:
-        """Triggered by config entry options updates."""
-        if hass.data[DOMAIN].event_unsub:
-            hass.data[DOMAIN].event_unsub()
-        await hass.data[DOMAIN].async_update()
+async def async_options_updated(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Triggered by config entry options updates."""
+    coordinator: IslamicPrayerDataUpdateCoordinator = hass.data[DOMAIN]
+    if coordinator.event_unsub:
+        coordinator.event_unsub()
+    await coordinator.async_request_refresh()
```

Comment:
Any suggestions on how to test this. I already have a test for updating options in test_config_flow.

Reply:
Hello @epenet , appreciate your advise on this.

Reply:
I'm sorry but I cannot help you with this. Maybe someone more knowledgeable than me can take a look...

Reply:
I got working by doing `hass.config_entries.async_setup` under the options test.